web/src/index.tsx

```diff
@@ -60,38 +62,52 @@ getAppWs().then(async (client) => {
     const profilesInfo = await client.appInfo({
       installed_app_id: MAIN_APP_ID,
     })
-    const { cell_id: cellId } = profilesInfo.cell_data.find(
-      ({ role_name }) => role_name === PROFILES_ROLE_NAME
-    )
-    const [_dnaHash, agentPubKey] = cellId
+
+    console.log(profilesInfo.cell_info)
+    const cellInfo = profilesInfo.cell_info[PROFILES_ROLE_NAME][0]
+    const cellId = ("Provisioned" in cellInfo) ? cellInfo.Provisioned.cell_id : undefined
+    console.log('cell_id', cellId)
+
+    console.log('before authorize signing key')
+    // authorize zome call signer
+    const adminWs = await getAdminWs()
+    await authorizeNewSigningKeyPair(adminWs, cellId!, profilesZomeFunctions);
+    console.log('after authorize key')
+    
+
+    const [_dnaHash, agentPubKey] = cellId!
     // cache buffer version of agentPubKey
     setAgentPubKey(agentPubKey)
-    const cellIdString = cellIdToString(cellId)
+    const cellIdString = cellIdToString(cellId!)
```

Comment:
instead of using all these `!` let's throw an error in the case where cellId ends up being undefined cause nothing would work anyways.